### PR TITLE
Add ability to test multiple server configurations

### PR DESCRIPTION
**Pull request by drew-gross**
This is a little janky due to needing to blow away some global state on every server reinitialization, and not blow away other global state. So it's kinda a stop-gap solution.

Amount of time the tests take is not significantly impacted.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -54,20 +54,36 @@ addParseCloud();
...
+  clientKey = '',
+  javascriptKey = '',
+  dotNetKey = '',
+  restAPIKey = '',
+  fileKey = 'invalid-file-key',
+  facebookAppIds = [],
+  enableAnonymousUsers = true,
+  oauth = {},
+  serverURL,
+}) {
```
Comment (flovilmart) [inline src/index.js]: :+1:

**Comment by drew-gross** [comment]
Test failure seems unrelated. Open/close.

File: src/index.js (hunk 1 above)
Comment (drew-gross) [inline src/index.js]: Is that an accept? :D

Comment (flovilmart) [inline src/index.js]: I like that syntax, learning ES6 at the same time you know... js/node is only my part-time pleasure.

**Comment by flovilmart** [comment]
that will go well with #501  ;)

**Comment by drew-gross** [comment]
Yah I did this for the purpose of testing adapters easier.

**Comment by drew-gross** [comment]
Gonna skip review on this because it really only changes tests and I'm tired of fixing merge conflicts. Sorry @nlutsenko :(

**Comment by nlutsenko** [comment]
All good, I love the style changes here as well.!

**Pull request merged**